include/swift/RemoteInspection/TypeRefBuilder.h

```diff
@@ -1043,6 +1052,30 @@ class TypeRefBuilder {
     ReflectionInfos.push_back(I);
     auto InfoID = ReflectionInfos.size() - 1;
     assert(InfoID <= UINT32_MAX && "ReflectionInfo ID overflow");
+
+    for (auto BuiltinTypeDescriptor : I.Builtin) {
```

Comment:
It might be good to make this lazy. I don't think `leaks` et al will care, but `swift-inspect` has paths that would never use this info and might benefit. We don't have to do that now, though.

Reply:
I'm not sure if anything actually relies on this, but we'd want to keep the cache up to date when new reflection infos are added. I suggest tracking the index of the last reflection info that's been cached, like:

```
uint32_t NormalizedReflectionNameCacheLastReflectionInfoCached;
```

Then in `getBuiltinTypeInfo`:

```
for(; NormalizedReflectionNameCacheLastReflectionInfoCached < ReflectionInfos.size(); NormalizedReflectionNameCacheLastReflectionInfoCached++)
  ...cache ReflectionInfos[NormalizedReflectionNameCacheLastReflectionInfoCached]...
  ...